homeassistant/components/withings/sensor.py

```diff
@@ -0,0 +1,427 @@
+    ),
+    WithingsSleepSummaryAttribute(
+        const.MEAS_SLEEP_HEART_RATE_MAX,
+        const.MEASURE_TYPE_SLEEP_HEART_RATE_MAX,
+        'Maximum heart rate', const.UOM_BEATS_PER_MINUTE, 'mdi:heart-pulse'
+    ),
+    WithingsSleepSummaryAttribute(
+        const.MEAS_SLEEP_RESPIRATORY_RATE_AVERAGE,
+        const.MEASURE_TYPE_SLEEP_RESPIRATORY_RATE_AVERAGE,
+        'Average respiratory rate', const.UOM_BREATHS_PER_MINUTE, None
+    ),
+    WithingsSleepSummaryAttribute(
+        const.MEAS_SLEEP_RESPIRATORY_RATE_MIN,
+        const.MEASURE_TYPE_SLEEP_RESPIRATORY_RATE_MIN,
+        'Minimum respiratory rate', const.UOM_BREATHS_PER_MINUTE, None
+    ),
+    WithingsSleepSummaryAttribute(
+        const.MEAS_SLEEP_RESPIRATORY_RATE_MAX,
+        const.MEASURE_TYPE_SLEEP_RESPIRATORY_RATE_MAX,
+        'Maximum respiratory rate', const.UOM_BREATHS_PER_MINUTE, None
+    ),
+]
+
+WITHINGS_MEASUREMENTS_MAP = {
+    attr.measurement: attr for attr in WITHINGS_ATTRIBUTES
+}
+
+
+class WithingsHealthSensor(Entity):
+    """Implementation of a Withings sensor."""
+
+    def __init__(self,
+                 data_manager: WithingsDataManager,
+                 attribute: WithingsAttribute) -> None:
+        """Initialize the Withings sensor."""
+        self._data_manager = data_manager
+        self._attribute = attribute
+        self._state = None
+
+        self._slug = self._data_manager.slug
+        self._user_id = self._data_manager.api.get_credentials().user_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return 'Withings {} {}'.format(self._attribute.measurement, self._slug)
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return 'withings_{}_{}_{}'.format(
+            self._slug,
+            self._user_id,
+            slugify(self._attribute.measurement)
+        )
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity, if any."""
+        return self._attribute.unit_of_measurement
+
+    @property
+    def icon(self) -> str:
+        """Icon to use in the frontend, if any."""
+        return self._attribute.icon
+
+    @property
+    def device_state_attributes(self):
+        """Get withings attributes."""
+        return self._attribute.__dict__
+
+    async def async_update(self) -> None:
+        """Update the data."""
+        _LOGGER.debug(
+            "Async update slug: %s, measurement: %s, user_id: %s",
+            self._slug, self._attribute.measurement, self._user_id
+        )
+
+        if isinstance(self._attribute, WithingsMeasureAttribute):
+            _LOGGER.debug("Updating measures state")
+            self._data_manager.update_measures()
```

Comment:
The nokia library doesn't support asyncio, it seems. Is this call doing I/O? We're not allowed to do I/O in the event loop.